### PR TITLE
TransactionBuilder: avoid extra getPublicKeyBuffer calls

### PR DESCRIPTION
**Pull request by dcousens**
Not sure how we missed this one! Anyway,  quite the performance increase.

### CODE REVIEW

File: src/transaction_builder.js [hunk 1]
```diff
@@ -140,6 +140,36 @@ function extractInput (transaction, txIn, vin) {
...
+        scriptType: scriptType
+      }
+
+    case 'multisig':
+      return {
+        pubKeys: outputScriptChunks.slice(1, -2),
+        scriptType: scriptType
+      }
+  }
+}
```
Comment (dcousens) [inline src/transaction_builder.js]: Also moved the function higher up,  with the rest of these convenience functions

**Comment by dcousens** [comment]
ping @fanatid for post merge ACK

File: src/transaction_builder.js [hunk 2]
```diff
@@ -465,7 +466,7 @@ TransactionBuilder.prototype.sign = function (index, keyPair, redeemScript, hash
 
       // if we don't have a scriptType, assume pubKeyHash otherwise
       if (!input.scriptType) {
-        input.prevOutScript = bscript.pubKeyHashOutput(bcrypto.hash160(keyPair.getPublicKeyBuffer()))
+        input.prevOutScript = bscript.pubKeyHashOutput(bcrypto.hash160(kpPubKey))
         input.prevOutType = 'pubkeyhash'
```
Comment (dcousens) [inline src/transaction_builder.js]: This will be a significant increase for `.sign` in the P2PKH case... @jprichardson lol

**Comment by fanatid** [comment]
post ACK

**Pull request merged**